src/wasm-type.h

```diff
@@ -48,6 +50,13 @@ class Type {
   // But converting raw uint32_t is more dangerous, so make it explicit
   constexpr explicit Type(uint32_t id) : id(id){};
 
+  explicit Type(const std::vector<ValueType>& types);
+
+  size_t getNumValueTypes() const;
+  const std::vector<ValueType> getValueTypes() const;
+
+  bool isValueType() { return id >= i32 && id < _last_value_type; }
```

Comment:
Isn't this check the same as `isConcreteType`?

Reply:
No, `isConcreteType` includes multivalue types as well.

Reply:
I see, thanks.

How about `isSingleType`? "value" doesn't seem clear enough to me.

(Also, may as well be `isSingle`, as the whole thing is a type?)